modules/Microsoft.EventGrid/systemTopics/eventSubscriptions/deploy.bicep

```
@@ -0,0 +1,90 @@
+param destination object
+
+@description('Optional. The event delivery schema for the event subscription.')
+@allowed(
+  [
+    'CloudEventSchemaV1_0'
+    'CustomInputSchema'
+    'EventGridSchema'
+    'EventGridEvent'
+  ]
+)
+param eventDeliverySchema string = 'EventGridSchema'
+
+@description('Optional. The expiration time for the event subscription. Format is ISO-8601 (yyyy-MM-ddTHH:mm:ssZ).')
+param expirationTimeUtc string = ''
+
+@description('Optional. The filter for the event subscription. (See https://learn.microsoft.com/en-us/azure/templates/microsoft.eventgrid/eventsubscriptions?pivots=deployment-language-bicep#eventsubscriptionfilter for more information).')
+param filter object = {}
+
+@description('Optional. The list of user defined labels.')
+param labels array = []
+
+@description('Optional. The retry policy for events. This can be used to configure the TTL and maximum number of delivery attempts and time to live for events.')
+param retryPolicy object = {}
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name, location)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource eventGridSystemTopic 'Microsoft.EventGrid/systemTopics@2022-06-15' existing = {
+  name: eventGridSystemTopicName
+}
+
+resource eventSubscription 'Microsoft.EventGrid/systemTopics/eventSubscriptions@2022-06-15' = {
+  name: name
+  parent: eventGridSystemTopic
+  properties: {
+    deadLetterDestination: !empty(deadLetterDestination) ? deadLetterDestination : null
+    deadLetterWithResourceIdentity: !empty(deadLetterWithResourceIdentity) ? deadLetterWithResourceIdentity : null
+    deliveryWithResourceIdentity: !empty(deliveryWithResourceIdentity) ? deliveryWithResourceIdentity : null
+    destination: destination
+    eventDeliverySchema: eventDeliverySchema
+    expirationTimeUtc: !empty(expirationTimeUtc) ? expirationTimeUtc : ''
+    filter: !empty(filter) ? filter : {}
+    labels: !empty(labels) ? labels : []
+    retryPolicy: !empty(retryPolicy) ? retryPolicy : null
+  }
+}
+
+@description('The name of the event grid topic.')
+output name string = eventSubscription.name
+
+@description('The resource ID of the event grid topic.')
+output resourceId string = eventSubscription.id
+
+@description('The name of the resource group the event grid topic was deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The location the resource was deployed into.')
+output location string = eventGridSystemTopic.location
```

```suggestion
output location string = eventSubscription.location
```
Or does only the event subscription have a location?